torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -587,10 +586,17 @@ bool ProcessGroupNCCL::WorkNCCL::startedGPUExecutionInternal() const {
   if (!timingEnabled_) {
     return false;
   }
+  auto mode = cudaStreamCaptureModeThreadLocal;
+  // push
+  cudaThreadExchangeStreamCaptureMode(&mode);
   // Checking the work's corresponding CUDA event's status
   if (!ncclStartEvent_->query()) {
+    // pop
+    cudaThreadExchangeStreamCaptureMode(&mode);
     return false;
   }
+  // pop
+  cudaThreadExchangeStreamCaptureMode(&mode);
```

Comment:
Huh, this is interesting. We can change the setting only on the "offender" side and still expect things to work? If so, that's great. (won't mess with user's setting should there be any).

nit: can we consolidate the two pops? Thanks!

Reply:
Q: does `cudaThreadExchangeStreamCaptureMode` need a correct device context? (I don't know)